src/welcome-bare-metal.md

```diff
@@ -0,0 +1,32 @@
+# Welcome to Bare Metal Rust
+
+This is a standalone one-day course about bare-metal Rust, aimed at people who are familiar with the
+basics of Rust (perhaps from completing the Comprehensive Rust course), and ideally also have some
+experience with bare-metal programming in some other language such as C.
+
+Today we will talk about 'bare-metal' Rust: running Rust code without an OS underneath us. This will
+be divided into several parts:
+
+ * What is `no_std` Rust?
+ * Writing firmware for microcontrollers.
+ * Writing bootloader / kernel code for application processors.
+ * Some useful crates for bare-metal Rust development.
+
+For the microcontroller part of the course we will use the [BBC micro:bit](https://microbit.org/) v2
+as an example. It's a [development board](https://tech.microbit.org/hardware/) based on the Nordic
+nRF51822 microcontroller with some LEDs and buttons, an I2C-connected accelerometer and compass, and
+an on-board SWD debugger.
+
+To get started, install some tools we'll need later:
+
+```bash
+sudo apt install gdb-multiarch picocom
+rustup update
```

Comment:
I think this line should go: it's not our job to tell people to update `rustup` (just like we don't have an `apt update` above:

```suggestion
```

Reply:
Updating the toolchain is the important point here. Some of the examples depend on a recent rustc version, and I'd rather not have to go around helping people debug confusing error messages due to trying to build with an out of date compiler.

Reply:
I see, in that case, I would suggest writing

```suggestion
rustup default 1.65.0 // or later
```

to ensure they use a known version. When I first ran these commands, I was on Rust 1.59.0 since I was testing something with that version — all the `rustup update` commands in the world wouldn't have helped me :smile: